src/integration/billing/stripe/StripeBillingIntegration.ts

```diff
@@ -694,14 +691,14 @@ export default class StripeBillingIntegration extends BillingIntegration<StripeB
     };
   }
 
-  private checkIfStripeIsInitialized() {
-    if (!this.stripe) {
-      throw new BackendError({
-        source: Constants.CENTRAL_SERVER,
-        module: MODULE_NAME, method: 'checkIfStripeIsInitialized',
-        action: ServerAction.CHECK_CONNECTION,
-        message: 'No connection to Stripe available'
-      });
+  private async initializeStripe() {
+    this.settings.currency = this.stripeSettings.currency;
+    // Get Crypto Setting
+    const cryptoSetting = await Cypher.getCryptoSetting(this.tenantID);
+    if (this.settings.secretKey) {
```

Comment:
I would prefer that you trap the missing key a lower level and throw an error. It's a requirement to have a key set at the tenant level for e-mobility to work.